### DISCUSSION TITLE
Geckos.io vs dgram

### DISCUSSION THREAD

**Discussion by jebarpg** [geckosio/geckos.io#264] [General]
What is the difference between using geckos.io and dgram for udp sockets?

const dgram = require('dgram');
const socket = dgram.createSocket('udp4');

**Comment by eltonnobrega** [comment]
geckos.io was made to help developers who want to make multiplayer games for browsers.
Browsers do not support pure UDP ('dgram'). But you can use it for non-browser games.
Geckos.io uses datachannel currently and in the future, maybe webtransport.

I liked how socket.io tries to upgrade the connection to try to use the protocol better. https://socket.io/get-started/webtransport
